iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/CommandDetector.java

```diff
@@ -78,44 +82,149 @@ public List<Command> search(List<ArgumentResolution> resolvedArguments) {
    */
   // Cognitive Complexity of methods should not be too high
   @SuppressWarnings("java:S3776")
+  // TODO: Rework commentary
   private List<ArgumentResolution> fullMatch(Deque<ArgumentResolution> argumentStack) {
-    List<ArgumentResolution> commandArguments = new ArrayList<>();
-    Deque<CommandPredicate> predicateStack = new LinkedList<>(predicates);
-    while (!predicateStack.isEmpty()) {
-      CommandPredicate currentPredicate = predicateStack.pollFirst();
-      ArgumentResolution resolution = argumentStack.pollFirst();
+    CommandDetectionStatus detectionStatus = new CommandDetectionStatus(argumentStack, new LinkedList<>(predicates), new ArrayList<>());
+    while (!detectionStatus.predicateStack.isEmpty()) {
+
+      detectionStatus.setCurrentPredicate(detectionStatus.predicateStack.pollFirst());
```

Comment:
This method does not need a parameter as the only provided argument is of the object itself